src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -40,29 +34,60 @@ import org.gradle.kotlin.dsl.withType
 @Suppress("UnstableApiUsage")
 class NexusPublishPlugin : Plugin<Project> {
 
-    companion object {
-        private val serverUrlToStagingRepoUrl = ConcurrentHashMap<URI, URI>()
-    }
-
     override fun apply(project: Project) {
-        project.pluginManager.apply(MavenPublishPlugin::class)
+        require(project == project.rootProject) {
+            "Plugin must be applied to the root project"
```

Comment:
It would probably make sense to add the actual project path to the error message.

Reply:
Done.